frontend/app/src/App.tsx

```diff
@@ -639,11 +642,16 @@ export class App extends PureComponent<Props, State> {
       //   1. this is the first time we establish a websocket connection to the
       //      server, or
       //   2. our last script run attempt was interrupted by the websocket
-      //      connection dropping.
-      if (!this.sessionInfo.last || lastRunWasInterrupted) {
+      //      connection dropping, or
+      //   3. the host explicitly requested a reconnect
+      if (
+        !this.sessionInfo.last ||
+        lastRunWasInterrupted ||
+        this.state.reconnectRequested
+      ) {
```

Comment:
My initial thought is hesitation for another state variable here. If anything, this seems tied to the connection package and we should perhaps have that handled in some way, but I see how that creates more efforts.

Have we thought about leveraging `ScriptRunState`? Where we set it to `RERUN_REQUESTED` on restartWebsocketConnection which will make `lastRunWasInterrupted` `true` and deliver the outcome you want (the ScriptRunState should reset when the SessionStatus changes from the script run...actually running). We'd change the meaning of `lastRunWasInterrupted` to something like `wasRerunRequested` and add a comment explaining the fact that RUNNING is used to determine if the last one was interrupted.

Reply:
In reality, we could potentially set scriptRunState to RERUN_REQUESTED if at the point of disconnect we see the script run state as still RUNNING.